src/app.jsx

```diff
@@ -30,18 +29,78 @@ import { NetworkList } from "./components/networks/networkList.jsx";
 import { VmDetailsPage } from './components/vm/vmDetailsPage.jsx';
 import { CreateVmAction } from "./components/create-vm-dialog/createVmDialog.jsx";
 import LibvirtSlate from "./components/libvirtSlate.jsx";
-import { isObjectEmpty, dummyVmsFilter, vmId } from "./helpers.js";
+import { dummyVmsFilter, vmId } from "./helpers.js";
 import { InlineNotification } from 'cockpit-components-inline-notification.jsx';
 import {
+    getApiData,
     usageStartPolling,
     usageStopPolling,
 } from "./libvirt-dbus.js";
+import { getOsInfoList, getLoggedInUser } from './libvirt-common.js';
+import { useObject, useEvent } from "hooks";
+import * as service from 'service.js';
+import store from './store.js';
 
+import { updateLibvirtState } from './actions/store-actions.js';
 const _ = cockpit.gettext;
 
 superuser.reload_page_on_change();
 
-class App extends React.Component {
+export const App = ({ name }) => {
+    const libvirtService = useObject(() => service.proxy(name), null, []);
+    const prevLibvirtServiceRef = useRef();
+    const [loadingResources, setLoadingResources] = useState(false);
+    const [error, setError] = useState('');
+
+    useEvent(libvirtService, "changed", () => {
+        store.dispatch(updateLibvirtState({
+            activeState: libvirtService.state,
+        }));
+        if (libvirtService.state == 'failed') {
+            console.warn("Libvirtd has failed");
```

Comment:
Thanks! If that helps to write a more robust naughty pattern, then by all means let's do so. I.e. keep the current one and additionally match on this line.

Reply:
i am waiting for test results to check :)